pkg/pq/priority_queue.go

```diff
@@ -21,43 +21,43 @@ import (
 )
 
 // PriorityQueue is a priority queue implemented with max heap.
-type PriorityQueue struct {
-	queue *internalQueue
+type PriorityQueue[V comparableValue] struct {
+	queue *internalQueue[V]
 }
 
 // NewPriorityQueue creates an instance of NewPriorityQueue.
-func NewPriorityQueue() *PriorityQueue {
-	pq := &internalQueue{}
+func NewPriorityQueue[V comparableValue]() *PriorityQueue[V] {
+	pq := &internalQueue[V]{}
 	heap.Init(pq)
 
-	return &PriorityQueue{
+	return &PriorityQueue[V]{
 		queue: pq,
 	}
 }
 
 // Peek returns the maximum element from this PriorityQueue.
-func (pq *PriorityQueue) Peek() Value {
-	return pq.queue.Peek().(*pqItem).value
+func (pq *PriorityQueue[V]) Peek() V {
+	return pq.queue.Peek().(*pqItem[V]).value
 }
 
 // Pop removes and returns the maximum element from this PriorityQueue.
-func (pq *PriorityQueue) Pop() Value {
-	return heap.Pop(pq.queue).(*pqItem).value
+func (pq *PriorityQueue[V]) Pop() V {
+	return heap.Pop(pq.queue).(*pqItem[V]).value
 }
 
 // Push pushes the element x onto this PriorityQueue.
-func (pq *PriorityQueue) Push(value Value) {
+func (pq *PriorityQueue[V]) Push(value V) {
 	item := newPQItem(value)
 	heap.Push(pq.queue, item)
 }
 
 // Len is the number of elements in this PriorityQueue.
-func (pq *PriorityQueue) Len() int {
+func (pq *PriorityQueue[V]) Len() int {
 	return pq.queue.Len()
 }
 
 // Release deletes the given value from this PriorityQueue.
-func (pq *PriorityQueue) Release(value Value) {
+func (pq *PriorityQueue[V]) Release(value V) {
 	idx := -1
 	for i, item := range *pq.queue {
 		if item.value == value {
```

Comment:
It looks like we have the wrong code on this line. 

If all values ​​in the struct are comparable, we can still use `==`, but this can lead to mistakes. So we have to use `Less()` to check for equality.

```suggestion
		if !item.value.Less(value) && !value.Less(item.value) {
```

After changing this code we can use `Value` directly as a type parameter.

Reply:
It was a problem I hadn't thought of. I learned a good idea that using 'Less' in both ways can replace 'equal'. The type comparableValue which was created because of the '==' operation could also be removed. Thank you!